### PR TITLE
implement heatmap #187

### PR DESCRIPTION
**Pull request by egaxhaj-figure**
## Description

* Heatmap API endpoint `/api/v2/txs/heatmap` for when Txs are submitted during the day.

closes: #187 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/provenance-io/explorer/blob/main/CONTRIBUTING.md#pr-targeting))
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote unit and integration
- [x] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: service/src/main/kotlin/io/provenance/explorer/domain/entities/Blocks.kt [hunk 1]
```diff
@@ -286,6 +286,21 @@ class BlockCacheHourlyTxCountsRecord(id: EntityID<DateTime>) : Entity<DateTime>(
             }
         }
 
+        fun getTxHeatmap() = transaction {
+            val blockTimestamp = BlockCacheHourlyTxCountsTable.blockTimestamp
+            val txCount = BlockCacheHourlyTxCountsTable.txCount
+            BlockCacheHourlyTxCountsTable
+                .slice(blockTimestamp, txCount)
```
Comment (crussell-figure) [inline service/src/main/kotlin/io/provenance/explorer/domain/entities/Blocks.kt:293]: This query needs to be summed up to the hour level.

**Review by crussell-figure — CHANGES_REQUESTED** [review]

File: service/src/main/kotlin/io/provenance/explorer/domain/entities/Blocks.kt (hunk 1 above)
Comment (egaxhaj-figure) [reply service/src/main/kotlin/io/provenance/explorer/domain/entities/Blocks.kt:293]: It's using `block_cache_hourly_tx_counts` table that is used by `Tx History`. The table is already summed up to the hour level.

The only difference between `heatmap` and `tx history` is that we don't set limits on `heatmap`. Should there be more slicing to the `heatmap` results?

Comment (crussell-figure) [reply service/src/main/kotlin/io/provenance/explorer/domain/entities/Blocks.kt:293]: Brain fart. You are correct. :nothing-to-see-here:

Comment (crussell-figure) [reply service/src/main/kotlin/io/provenance/explorer/domain/entities/Blocks.kt:293]: So you have it returning a list of hourly records, but they need to be summed across day of the week. Brain fart on the hourly trunc already, but thats only half of it.

Comment (egaxhaj-figure) [reply service/src/main/kotlin/io/provenance/explorer/domain/entities/Blocks.kt:317]: Added `DAY` granularity

**Review by crussell-figure — APPROVED** [review]
Looks great!

**Pull request merged**